app/src/main/java/com/my/journeyplanner/presenters/MainPresenter.kt

```diff
@@ -20,30 +24,91 @@ class MainPresenter(private val view: MainContract.View) : MainContract.Presente
     }
 
     override fun onPlanMyJourneyClicked() {
-        val apiService = JourneyPlannerApiService.createApiService()
-
-        call = apiService.getJourneyResults(
+        call = JourneyPlannerApiService.createApiService().getJourneyResults(
             view.getFromLocation().text.toString(),
             view.getToLocation().text.toString()
         )
 
-/*            TODO("Default adapter treats response code 300 Multiple Options as unsuccessful and")
-            TODO("puts response in the errorBody but is actually a valid response.")
-            TODO("Implement custom adapter to parse the valid response in the errorBody")   */
-        call!!.enqueue(object : Callback<JourneyPlannerDisambiguationResult> {
-            override fun onFailure(call: Call<JourneyPlannerDisambiguationResult>, throwable: Throwable) {
-                Log.e(TAG, "Other error: ${throwable.message}")
-                Log.e(TAG, "Other error: ${throwable.stackTrace.forEach { println(it) }}")
+        call!!.enqueue(object : Callback<JourneyPlanner.ItineraryResult> {
+            override fun onFailure(
+                call: Call<JourneyPlanner.ItineraryResult>,
+                throwable: Throwable
+            ) {
+                if (throwable is IOException) {
+                    Log.d(TAG, "*** Network failure ***")
+                } else {
+                    Log.d(TAG, "*** Conversion issue ***")
+                }
             }
 
             override fun onResponse(
-                call: Call<JourneyPlannerDisambiguationResult>,
-                response: Response<JourneyPlannerDisambiguationResult>
+                call: Call<JourneyPlanner.ItineraryResult>,
+                response: Response<JourneyPlanner.ItineraryResult>
             ) {
-                Log.i(TAG, "response: $response")
-                response.errorBody()?.apply {
-                    Log.i(TAG, "errorBody().source(): ${source()}")
-                    Log.i(TAG, "errorBody().charStream().readText(): ${charStream().readText()}")
+                if (response.isSuccessful) {
+                    Log.d(TAG, "*** ItineraryResult: ${response.body()} ***")
+//                TODO("Display data in new activity")
+                } else when (response.code()) {
+                    300 -> {
```

Comment:
didn't we make sure this doesn't happen?

Reply:
Yes, but I removed the MultiOptionResponseCodeInterceptor() as I couldn't get it to work with:
public @Nullable Converter<ResponseBody, ?> responseBodyConverter(Type type,
        Annotation[] annotations, Retrofit retrofit)
due to the generic return type ResponseBody i.e. I can't do this:
                                val converter: Converter<T, JourneyPlanner.NotIdentifiedResult> =
                                    JourneyPlannerApiService.fakeRetrofit.responseBodyConverter(
                                        JourneyPlanner.NotIdentifiedResult::class.java,
                                        arrayOfNulls<Annotation>(0)
                                    )
  converter.convert(response.body())
where:
  public @Nullable T body() {}
There'll be an obvious way but I haven't found it

Reply:
yeah I'll help you out on this later

Reply:
Reinstated MultiOptionResponseCodeInterceptor()